src/main/java/randoop/generation/InputSequenceSelector.java

```diff
@@ -0,0 +1,29 @@
+package randoop.generation;
+
+import java.util.Set;
+import randoop.sequence.ExecutableSequence;
+import randoop.sequence.Sequence;
+import randoop.util.SimpleList;
+
+/** Interface for selecting sequences as input for creating new sequences. */
+public interface InputSequenceSelector {
+  /**
+   * Choose a sequence used as input for creating a new sequence.
+   *
+   * @param candidates sequences to choose from
+   * @return the chosen sequence
+   */
+  Sequence selectInputSequence(SimpleList<Sequence> candidates);
+
+  /**
+   * Each {@link Sequence} within {@code inputSequences} is a subsequence of {@code eSeq}. At this
+   * point, the given {@link ExecutableSequence} has been executed and contains the execution time
+   * of the sequence as a whole. Make use of the execution time value that is stored within eSeq and
+   * associate it with each of the input sequences.
```

Comment:
Instead of relying on eSeq's execution time, we are now computing the execution time of each input sequence. Documentation updated to state:

The Executable Sequence eSeq also contains an ExecutionOutcome for each statement within the entire sequence. The ExecutionOutcome contains the execution time for a given statement. This information can be used to compute the execution time of any subsequence within eSeq.